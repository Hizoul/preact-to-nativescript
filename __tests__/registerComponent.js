import * as preactToNativeScript from "../index"

const h = preactToNativeScript.Preact.h

class MyComp {
  constructor() {
    this._eventListener = []
  }
  addChild() {
  }
  removeChild() {
  }
  _removeView() {
    
  }
  addEventListener(name, func, node) {
    this._eventListener[name] = func
  }
  removeEventListener(name, func, node) {
    this._eventListener[name] = null
  }
}

test("SegmentedBar", () => {
  const vnode = h("page", null, [
    h("stackLayout", null, [
      h("gradient")
    ])
  ])
  try {
  expect(preactToNativeScript.render(vnode)).toMatchSnapshot("before registered gradient")
  } catch (e) {
    expect(e).toMatchSnapshot("err")
  }
  preactToNativeScript.registerComponent("gradient", MyComp)
  expect(preactToNativeScript.render(vnode)).toMatchSnapshot("after registered gradient")
})
