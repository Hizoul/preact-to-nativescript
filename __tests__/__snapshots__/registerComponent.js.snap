// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedBar: after registered gradient 1`] = `
Page {
  "__n": "page",
  "__preactattr_": Object {},
  "_eventListener": Array [],
  "attributes": Array [],
  "childNodes": Array [
    StackLayout {
      "__n": "stackLayout",
      "__preactattr_": Object {},
      "_eventListener": Array [],
      "attributes": Array [],
      "childNodes": Array [
        MyComp {
          "__n": "gradient",
          "__preactattr_": Object {},
          "_eventListener": Array [],
          "attributes": Array [],
          "childNodes": Array [],
          "firstLoad": true,
          "loaded": false,
          "nodeName": "GRADIENT",
          "nodeType": 1,
          "parentNode": [Circular],
          "set": [Function],
        },
      ],
      "firstLoad": true,
      "loaded": false,
      "nodeName": "STACKLAYOUT",
      "nodeType": 1,
      "parentNode": [Circular],
      "set": [Function],
    },
  ],
  "content": StackLayout {
    "__n": "stackLayout",
    "__preactattr_": Object {},
    "_eventListener": Array [],
    "attributes": Array [],
    "childNodes": Array [
      MyComp {
        "__n": "gradient",
        "__preactattr_": Object {},
        "_eventListener": Array [],
        "attributes": Array [],
        "childNodes": Array [],
        "firstLoad": true,
        "loaded": false,
        "nodeName": "GRADIENT",
        "nodeType": 1,
        "parentNode": [Circular],
        "set": [Function],
      },
    ],
    "firstLoad": true,
    "loaded": false,
    "nodeName": "STACKLAYOUT",
    "nodeType": 1,
    "parentNode": [Circular],
    "set": [Function],
  },
  "firstLoad": true,
  "loaded": false,
  "mergeInto": undefined,
  "nodeName": "PAGE",
  "nodeType": 1,
  "parentNode": Object {
    "_removeView": [Function],
    "appendChild": [Function],
    "attributes": Array [],
    "childNodes": Array [
      [Circular],
    ],
    "mergeInto": undefined,
    "nodeName": "artificalParent",
    "nodeType": "artificalParent",
    "remove": [Function],
    "removeChild": [Function],
    "renderedComponent": VNode {
      "attributes": undefined,
      "children": Array [
        VNode {
          "attributes": undefined,
          "children": Array [
            VNode {
              "attributes": undefined,
              "children": Array [],
              "key": undefined,
              "nodeName": "gradient",
            },
          ],
          "key": undefined,
          "nodeName": "stackLayout",
        },
      ],
      "key": undefined,
      "nodeName": "page",
    },
    "renderedInto": null,
  },
  "renderedComponent": VNode {
    "attributes": undefined,
    "children": Array [
      VNode {
        "attributes": undefined,
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [],
            "key": undefined,
            "nodeName": "gradient",
          },
        ],
        "key": undefined,
        "nodeName": "stackLayout",
      },
    ],
    "key": undefined,
    "nodeName": "page",
  },
  "renderedInto": [Circular],
  "set": [Function],
}
`;

exports[`SegmentedBar: err 1`] = `
[Error: [31m[1mConfiguration error[22m:[39m
[31m[39m
[31mCould not locate module [1mtns-core-modules/ui/gradient[22m (mapped as [1mpreact-nativescript-mock/ui/gradient[22m)[39m
[31m[39m
[31mPlease check:[39m
[31m[39m
[31m"moduleNameMapper": {[39m
[31m  "/tns-core-modules(.*)/": "[1mpreact-nativescript-mock$1[22m"[39m
[31m},[39m
[31m"resolver": [1mundefined[22m[39m]
`;
